src/components/TextEditor/extensions/table/table-extension.ts

```diff
@@ -0,0 +1,275 @@
+import { Extension } from '@tiptap/core'
+import { VueRenderer } from '@tiptap/vue-3'
+import { Plugin, PluginKey } from 'prosemirror-state'
+import Table from '@tiptap/extension-table'
+import TableRow from '@tiptap/extension-table-row'
+import TableHeader from '@tiptap/extension-table-header'
+import TableCell from '@tiptap/extension-table-cell'
+import TableActionMenu from './TableActionMenu.vue'
+import TableCellActionHandle from './TableCellActionHandle.vue'
+import TableHandler from './TableHandler'
+
+export interface TableExtensionOptions {
+  scrollable: boolean
+  showActionHandles: boolean
+  minColumnWidth: number
+  maxColumnWidth: number
+}
+
+declare module '@tiptap/core' {
+  interface Commands<ReturnType> {
+    tableEditor: {
+      insertTable: (options: {
+        rows: number
+        cols: number
+        withHeaderRow?: boolean
+      }) => ReturnType
+      addColumnBefore: () => ReturnType
+      addColumnAfter: () => ReturnType
+      deleteColumn: () => ReturnType
+      addRowBefore: () => ReturnType
+      addRowAfter: () => ReturnType
+      deleteRow: () => ReturnType
+      deleteTable: () => ReturnType
+      toggleHeaderRow: () => ReturnType
+      toggleHeaderColumn: () => ReturnType
+    }
+  }
+}
+
+declare global {
+  interface HTMLElement {
+    tableHandler?: TableHandler
+  }
+}
+
+const TableMouseHandlerKey = new PluginKey('tableMouseHandler')
+
+export const TableExtension = Extension.create<TableExtensionOptions>({
+  name: 'Table',
+
+  addOptions() {
+    return {
+      scrollable: true,
+      showActionHandles: true,
+      minColumnWidth: 120,
+      maxColumnWidth: 300,
+    }
+  },
+
+  addExtensions() {
+    return [
+      Table.configure({
+        resizable: true,
+      }),
+      TableRow,
+      TableHeader,
+      TableCell,
+    ]
+  },
+
+  addCommands() {
+    return {
+      insertTable:
+        (options) =>
+        ({ commands }) => {
+          return commands.insertTable(options)
+        },
+      addColumnBefore:
+        () =>
+        ({ commands }) => {
+          return commands.addColumnBefore()
+        },
+      addColumnAfter:
+        () =>
+        ({ commands }) => {
+          return commands.addColumnAfter()
+        },
+      deleteColumn:
+        () =>
+        ({ commands }) => {
+          return commands.deleteColumn()
+        },
+      addRowBefore:
+        () =>
+        ({ commands }) => {
+          return commands.addRowBefore()
+        },
+      addRowAfter:
+        () =>
+        ({ commands }) => {
+          return commands.addRowAfter()
+        },
+      deleteRow:
+        () =>
+        ({ commands }) => {
+          return commands.deleteRow()
+        },
```

Comment:
are these even required? defined command is running the command of the same name?